src/main/java/net/fabricmc/installer/server/ServerInstaller.java

```diff
@@ -61,13 +62,35 @@ public class ServerInstaller {
 	public static final String DEFAULT_LAUNCH_JAR_NAME = "fabric-server-launch.jar";
 	private static final Pattern SIGNATURE_FILE_PATTERN = Pattern.compile("META-INF/[^/]+\\.(SF|DSA|RSA|EC)");
 
-	public static void install(Path dir, LoaderVersion loaderVersion, String gameVersion, InstallerProgress progress) throws IOException {
+	public static void install(Path dir
+			, LoaderVersion loaderVersion
+			, String gameVersion
+			, InstallerProgress progress) throws IOException {
+
+		Scanner sc = new Scanner(System.in);
 		Path launchJar = dir.resolve(DEFAULT_LAUNCH_JAR_NAME);
-		install(dir, loaderVersion, gameVersion, progress, launchJar);
+		String RESPONSE_TO_OPTION;
+		if (isSudo()) {
+			System.out.println("WARNING: Running this installer with sudo might cause issues with the application.");
+			System.out.print("Do you really want to continue? (y/n): ");
+			RESPONSE_TO_OPTION = sc.nextLine().trim(); // Use nextLine to capture the entire line
+
+			if (RESPONSE_TO_OPTION.equalsIgnoreCase("y")) {
+				install(dir, loaderVersion, gameVersion, progress, launchJar);
+			} else if (RESPONSE_TO_OPTION.equalsIgnoreCase("n")){
+				throw new RuntimeException("Installation aborted by the user.");
+			} else {
+				System.out.println("Invalid Response!!");
+			}
```

Comment:
Just assume the user aborted it. This will be removed anyway as a Scanner isnt the right solution